### ISSUE TITLE
Making the addition of `prov:generatedAtTime` and `prov:wasAttributedTo` optional

### ISSUE THREAD

**Issue by vemonet** [fair-workflows/nanopub#149]
The `nanopub` python library is automatically adding `prov:generatedAtTime` duplicated twice in the pubinfo and prov graph. And a `prov:wasAttributedTo` in the pubinfo. 

I added 2 parameters to the `Publication.from_assertion` method: `add_generated_at_time` (default to True) and `attribute_publication_to_profile` (default to True): https://github.com/vemonet/nanopub/commit/0df2db6203ab3679334449307cb797535d24693a 

So that the default behavior is still the same (adding those metadata), but users can also choose to add them by themselves

Is that something that you would be interested to see merged here? @svenvanderburg

**Comment by raar1** [comment]
That would be great @vemonet ! I agree that the duplication is not ideal and users should definitely be able to specify these times themselves. Would you be willing to open a PR with your suggested changes and we can review it? Thanks again for your help.

**Comment by raar1** [comment]
@vemonet Thanks again for this. I've had a look at the PR and it looks good. I completely agree that it should be possible to disable the time and profile attributions. What I'm less sure about is what the defaults should be.

Currently in the code if `attribute_publication_to_profile` is False, then `publication_attributed_to` will be ignored. This makes sense, of course, if the user has explicitly said they don't want to attribute the publication to a profile. However, if `attribute_publication_to_profile` is set False by default (which I think you were suggesting could be a better choice) then this might confuse users who do e.g.:
```python
publication = Publication.from_assertion(assertion_rdf=my_assertion, publication_attributed_to=someorcid)
```
and then wonder why `publication_attributed_to` isn't working.

I think either we keep your code as it is, but leave `attribute_publication_to_profile=True` as default. Or we make the use of `publication_attributed_to` override `attribute_publication_to_profile`, and always attribute if it is set.

What do you think?

**Comment by vemonet** [comment]
Hi @raar1, I agree we should leave those defaults. After second thought, having the ORCID of the publisher and the time the nanopub was published added to the pubinfo graph by default is a good practice. The only thing is that it can be a bit surprising when you are used to publish with the java tool that does not add anything

There is also the fact that `prov:generatedAtTime` is automatically added to the pubinfo and to the provenance graphs that feels a bit weird. It would make more sense if it would be only added to the pubinfo. 
But publishing a nanopub with nothing in the provenance graph will fail, so it can be useful to have at least the generatedAtTime in there by default...

**Comment by raar1** [comment]
Yes, I see your point... This library uses the java tool, after all, so users of that will naturally expect `nanopub` to do something similar. I suppose we considered the java tool to be 'lower level' in some way (in the sense that, user has more control/does more work), and thought people might expect a python library to do more things by default, but we might not have chosen the right way of doing it. Perhaps there could be a flag that turns off basically all options that add triples automatically, so you can use it more like the base java tool? I think it would still validate the rdf graph. Not sure if this is simpler though.

With regards to the `prov:generatedAtTime` essentially duplicated in `provenance` and `pubinfo` I agree completely, and I don't really like it either. At least your changes make it possible to override this behaviour, so it's already an improvement :)
